labgrid/util/proxy.py

```diff
@@ -39,15 +43,23 @@ def get_host_and_port(cls, res, *, default_port=None, force_port=None):
         """
         assert isinstance(res, Resource)
 
-        s = urlsplit('//'+res.host)
+        url = '//' + res.host
+        s = urlsplit(url)
         host = s.hostname
+
         if force_port:
             port = force_port
         elif s.port:
             port = s.port
         else:
             port = getattr(res, 'port', None) or default_port
 
+        if host is None:
+            raise ProxyError(f"Invalid url: {url} does not contain a host")
+
+        if port is None:
+            raise ProxyError(f"Invalid url: {url} does not contain a port and no default set")
+
```

Comment:
The input to this function is not an URL, so these messages are a bit confusing. The port can come from the arguments or from the resource. The hostname is always from the resource and can't be missing at this point (res.host should never be None, though).

Reply:
Hm, I don't think we need the check here than, I'll remove it if that is fine with you.

Reply:
Sounds fine.